core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/KinesisSink.java

```diff
@@ -0,0 +1,165 @@
+      Object fieldValue = input.get(field.getName());
+      String data = (fieldValue != null) ? fieldValue.toString() : NULL_STRING;
+      dataArray.add(data);
+    }
+    emitter.emit(new KeyValue<>(NullWritable.get(), new Text(Joiner.on(",").join(dataArray))));
+  }
+
+  private static class KinesisOutputFormatProvider implements OutputFormatProvider {
+
+    private final Map<String, String> conf;
+    private KinesisOutputFormatProvider(KinesisConfig config) {
+      this.conf = new HashMap<>();
+      conf.put(Properties.KinesisRealtimeSink.ACCESS_ID, config.awsAccessKey);
+      conf.put(Properties.KinesisRealtimeSink.ACCESS_KEY, config.awsAccessSecret);
+      conf.put(Properties.KinesisRealtimeSink.SHARD_COUNT, String.valueOf(config.getShardCount()));
+      conf.put(Properties.KinesisRealtimeSink.DISTRIBUTE, config.getDistribute());
+      conf.put(Properties.KinesisRealtimeSink.NAME, config.name);
+    }
+
+    @Override
+    public String getOutputFormatClassName() {
+      return KinesisOutputFormat.class.getName();
+    }
+
+    @Override
+    public Map<String, String> getOutputFormatConfiguration() {
+      return conf;
+    }
+  }
+
+  /**
+   * config file for Kinesis stream sink
+   */
+  public static class KinesisConfig extends ReferencePluginConfig {
+
+    @Description("The name of the Kinesis stream to output to. The stream will be created if it does not exist.")
+    private String name;
+
+    @Name(Properties.KinesisRealtimeSink.ACCESS_ID)
+    @Description("AWS access Id having access to Kinesis streams")
+    @Macro
+    private String awsAccessKey;
+
+    @Name(Properties.KinesisRealtimeSink.ACCESS_KEY)
+    @Description("AWS access key secret having access to Kinesis streams")
+    @Macro
+    private String awsAccessSecret;
+
+    @Name(Properties.KinesisRealtimeSink.SHARD_COUNT)
+    @Description("Number of shards to be created, each shard has throughput of 1mb/s")
+    @Macro
+    @Nullable
+    private Integer shardCount;
+
+    @Name(Properties.KinesisRealtimeSink.DISTRIBUTE)
+    @Description("Boolean to decide if the data has to be sent to a single shard or has to be uniformly distributed" +
+      "among all the shards. Default value is true")
+    private String distribute;
+
+    public KinesisConfig(String referenceName, String name, String awsAccessKey,
+                         String awsAccessSecret, Integer shardCount, String distribute) {
+      super(referenceName);
+      this.name = name;
+      this.awsAccessKey = awsAccessKey;
+      this.awsAccessSecret = awsAccessSecret;
+      this.shardCount = shardCount;
+      this.distribute = distribute;
+    }
+
+    public Integer getShardCount() {
+      return shardCount == null ? DEFAULT_SHARD_COUNT : shardCount;
+    }
+
+
+    public String getDistribute() {
+      //ensuring that distribute can have only 2 possible values "true" or "false". defaults to true
+      return String.valueOf("true".equals(distribute));
```

Comment:
I think its cleaner to do Boolean.toString(Boolean.parseBoolean(distribute)).
That way, you're not having the string "true" in your code.